### ISSUE TITLE
Cannot access port on Windows from Docker container via host-gateway

### ISSUE THREAD

**Issue by Morthy** [microsoft/WSL#12023]
### Windows Version

Microsoft Windows [Version 10.0.22631.4037]

### WSL Version

2.3.17.0

### Are you using WSL 1 or WSL 2?

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

Linux version 5.15.153.1-microsoft-standard-WSL2

### Distro Version

Ubuntu 20.04

### Other Software

Docker version 27.2.0, build 3ab4256

### Repro Steps

My setup is as follows.

WSL Config:

```[wsl2]
memory=25GB
networkingMode=mirrored
dnsTunneling=true
[experimental]
hostAddressLoopback=true
```

- Running a debugger listening on port 9003 on the Windows host
- Running docker engine directly on my WSL instance
- Running docker containers with a bridged network

I expect my docker containers to be able to connect to the debugger running on Windows, but they are not able to connect.

The docker container is configured with an extra_host to post host.docker.internal to the host gateway:

`      - "host.docker.internal:host-gateway"`

From the docker container, I get these results:

```
# telnet host.docker.internal 9003
Trying 10.253.0.1...
telnet: Unable to connect to remote host: Connection refused
```

If I try the IP of my Windows host directly, that does seem to work:

```
# telnet 192.168.178.84 9003
Trying 192.168.178.84...
Connected to 192.168.178.84.
Escape character is '^]'.
```

However, I would expect the first attempt to also work, so that I can avoid hardcoding my windows IP in my docker setup.

### Expected Behavior

Connecting to a Windows host port over the docker host IP or Ubuntu host IP should work

### Actual Behavior

Connection is not possible

### Diagnostic Logs

The networking logs were recorded while running both the unsuccessful and successful attempts above 

[WslNetworkingLogs-2024-09-11_18-52-21.zip](https://github.com/user-attachments/files/16966933/WslNetworkingLogs-2024-09-11_18-52-21.zip)

**Comment by zcobol** [comment]
`When set to True, will allow the Container to connect to the Host, or the Host to connect to the Container, by an IP address that's assigned to the Host`

Looks like you're connecting to the `IP address that's assigned to the Host` just fine!

**Comment by Morthy** [comment]
Yeah, that’s what is allowing the second test to work. I was under the expectation the first option would work too, unless im understanding wrong?

**Comment by Morthy** [comment]
So I'm not entirely sure why, but after a fresh boot today host.docker.internal is now resolving to the IP of my host, and so its working now.
